backend/services/prisonerProfileService.ts

```diff
@@ -98,26 +98,27 @@ export default ({
       ].map((apiCall) => logErrorAndContinue(apiCall))
     )
 
-    let hasDivergenceSupport = false
-    const hasSupportNeed = !(
-      neurodivergenceData === null ||
-      (neurodivergenceData[0]?.neurodivergenceSupport === undefined &&
-        neurodivergenceData[0]?.neurodivergenceSupport.length === undefined)
-    )
-    if (hasSupportNeed) {
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      const hasIdentifiedDivergenceSupportNeed = neurodivergenceData.some((element) => {
-        return element.neurodivergenceSupport === NeurodivergenceType.NoidentifiedNeurodiversityNeed
-      })
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      // eslint-disable-next-line @typescript-eslint/no-shadow
-      const hasRequiredSupport = neurodivergenceData.some((ele) => {
-        return ele.neurodivergenceSupport === NeurodivergenceType.NoidentifiedSupportRequired
-      })
-      hasDivergenceSupport = !(hasIdentifiedDivergenceSupportNeed || hasRequiredSupport)
+    const NeedNeurodivergenceSupport = (divergenceData) => {
```

Comment:
Sorry - maybe i wasn't clear, but this code needs to be outside the flow of the code - so in a different part of the file.

Doing it like this doesn't address my original comment